packages/blade/codemods/brand-refresh/transformers/index.ts

```diff
@@ -0,0 +1,161 @@
+  const newSource = file.source
+    .replace(
+      // gets both .50 and ['50'] or ["50"]
+      /theme\.typography\.fonts\.size\.?((\w+)|(\W.*\]))/g,
+      (originalString, match) => {
+        const token = match.replace(/\[|\]|'|"/g, '');
+        const replacement = fontSizeMap[token];
+        if (!replacement) {
+          return originalString;
+        }
+        return `${fontTokenPrefix}[${fontSizeMap[token]}]`;
+      },
+    )
+    .replace(
+      // gets both .50 and ['50'] or ["50"]
+      /theme\.typography\.lineHeights\.?((\w+)|(\W.*\]))/g,
+      (originalString, match) => {
+        const token = match.replace(/\[|\]|'|"/g, '');
+        const replacement = lineHeightMap[token];
+        if (!replacement) {
+          return originalString;
+        }
+        return `${lineHeightTokenPrefix}[${lineHeightMap[token]}]`;
+      },
+    );
+
+  const j = api.jscodeshift;
+  const root = j.withParser('tsx')(newSource);
+
+  // Select Typography elements based on their names
+  const typographyJSXElements = root
+    .find(j.JSXElement)
+    .filter((path) =>
+      ['Text', 'Title', 'Code', 'Display', 'Heading'].includes(path.value.openingElement.name.name),
+    );
+
+  // Replace Title with Heading and update the 'size' attribute
+  // <Title size="medium"> to <Heading size="xlarge">
+  typographyJSXElements
+    .filter((path) => path.value.openingElement.name.name === 'Title')
+    // replace with Heading
+    .replaceWith((path) => {
+      const { node } = path;
+
+      node.openingElement.name.name = 'Heading';
+      node.closingElement.name.name = 'Heading';
+
+      return node;
+    })
+    .find(j.JSXAttribute)
+    .filter((path) => path.node.name.name === 'size')
+    .replaceWith((path) => {
+      if (isExpression(path.node)) return path.node;
+
+      path.node.value.value = titleToHeadingMap[path.node.value.value] || 'large';
+
+      return path.node;
+    });
+
+  // Remove/Update the Title import from "@razorpay/blade/components"
+  root
+    .find(j.ImportDeclaration)
+    .filter((path) => path.value.source.value === '@razorpay/blade/components')
+    .find(j.ImportSpecifier)
+    .filter((path) => path.value.imported.name === 'Title')
+    .replaceWith((path) => {
+      // Check if Heading import is already present
+      const isHeadingImportPresent = path.parent.value.specifiers.some(
+        (node) => node.imported.name === 'Heading',
+      );
+      // // If Heading import is not present, update the "Title" import to use "Heading"
+      if (!isHeadingImportPresent) {
+        path.value.imported.name = 'Heading';
+      } else {
+        // If "Heading" import is present, remove the "Title" import
+        path.parent.value.specifiers = path.parent.value.specifiers.filter(
+          (node) => node.imported.name !== 'Title',
+        );
+      }
+
+      return path.node;
+    });
+
+  // Remove `type` prop from Typography Components
```

Comment:
In this when we remove `type` prop, we also have to make sure that `color` prop is set with same type.

```jsx
<Text type="subtle" >
```

Turns into -

```jsx
<Text color="surface.text.subtle.lowContrast">
```

Although we'll only be able to get the correct color token once we make color token changes.

Lets move this from typography to colors codemod when we do that? because if we just remove type prop things will break.

Reply:
Do we have this change for other values of `type` prop?